### PR TITLE
Updating examples

### PR DESCRIPTION
**Pull request by alexsu52**
### Changes

- Migrated to openvino.convert_model(...)
- Upgraded OpenVINO version to 2023.2 in PTQ TensorFlow example 

### Reason for changes

Aligned examples with the OpenVINO 2023.2

### Related tickets

125257

### Tests

test_examples 157

### CODE REVIEW

File: examples/post_training_quantization/tensorflow/mobilenet_v2/main.py [hunk 1]
```diff
@@ -146,16 +145,16 @@ def transform_fn(data_item):
...
 # Benchmark performance, calculate compression rate and validate accuracy
 
-ov_model = mo.convert_model(tf_model)
-ov_quantized_model = mo.convert_model(tf_quantized_model)
+ov_model = ov.convert_model(tf_model, share_weights=False)
+ov_quantized_model = ov.convert_model(tf_quantized_model, share_weights=False)
 
 fp32_ir_path = f"{ROOT}/mobilenet_v2_fp32.xml"
-ov.serialize(ov_model, fp32_ir_path)
+ov.save_model(ov_model, fp32_ir_path, compress_to_fp16=False)
```
Comment (KodiaqQ) [inline examples/post_training_quantization/tensorflow/mobilenet_v2/main.py:152]: Why the `compress_to_fp16` option was set to `False`?

File: examples/post_training_quantization/torch/mobilenet_v2/main.py [hunk 2]
```diff
@@ -138,22 +140,15 @@ def transform_fn(data_item: Tuple[torch.Tensor, int]) -> torch.Tensor:
...
 # (default: 300 samples) of the calibration dataset.
 
-calibration_dataset = nncf.Dataset(val_data_loader, transform_fn)
-torch_quantized_model = nncf.quantize(
-    torch_model,
-    calibration_dataset,
-    advanced_parameters=nncf.AdvancedQuantizationParameters(disable_bias_correction=True),
-)
+calibration_dataset = nncf.Dataset(val_data_loader, partial(transform_fn, device=device))
+torch_quantized_model = nncf.quantize(torch_model, calibration_dataset)
```
Comment (KodiaqQ) [inline examples/post_training_quantization/torch/mobilenet_v2/main.py:144]: What about accuracy numbers for the `test_examples` since bias correction was enabled?

File: examples/post_training_quantization/torch/mobilenet_v2/main.py [hunk 3]
```diff
@@ -138,22 +140,15 @@ def transform_fn(data_item: Tuple[torch.Tensor, int]) -> torch.Tensor:
...
 ###############################################################################
 # Benchmark performance, calculate compression rate and validate accuracy
 
 dummy_input = torch.randn(1, 3, 224, 224)
-ov_input_shape = (-1, 3, 224, 224)
-ov_model = mo.convert_model(torch_model.cpu(), example_input=dummy_input, input_shape=ov_input_shape)
-ov_quantized_model = mo.convert_model(
-    torch_quantized_model.cpu(), example_input=dummy_input, input_shape=ov_input_shape
-)
+ov_model = ov.convert_model(torch_model.cpu(), example_input=dummy_input)
```
Comment (KodiaqQ) [inline examples/post_training_quantization/torch/mobilenet_v2/main.py:150]: Why do we change the batch dimension from a dynamic to a static shape?

File: examples/post_training_quantization/tensorflow/mobilenet_v2/main.py [hunk 4]
```diff
@@ -146,16 +145,16 @@ def transform_fn(data_item):
 ###############################################################################
 # Benchmark performance, calculate compression rate and validate accuracy
 
-ov_model = mo.convert_model(tf_model)
-ov_quantized_model = mo.convert_model(tf_quantized_model)
+ov_model = ov.convert_model(tf_model, share_weights=False)
+ov_quantized_model = ov.convert_model(tf_quantized_model, share_weights=False)
```
Comment (KodiaqQ) [inline examples/post_training_quantization/tensorflow/mobilenet_v2/main.py:149]: Why the `share_weights` option was set to `False`?

File: examples/post_training_quantization/torch/ssd300_vgg16/main.py [hunk 5]
```diff
@@ -158,11 +157,11 @@ def main():
 
     fp32_onnx_path = f"{ROOT}/ssd300_vgg16_fp32.onnx"
     torch.onnx.export(model.cpu(), dummy_input, fp32_onnx_path)
-    ov_model = mo.convert_model(fp32_onnx_path)
+    ov_model = ov.convert_model(fp32_onnx_path)
```
Comment (KodiaqQ) [inline examples/post_training_quantization/torch/ssd300_vgg16/main.py:160]: Why do we still use the `PyTorch->ONNX->IR` conversion instead of the `PyTorch->IR` conversion as in `examples/post_training_quantization/torch/mobilenet_v2/main.py`?

File: examples/post_training_quantization/tensorflow/mobilenet_v2/main.py (hunk 1 above)
Comment (alexsu52) [reply examples/post_training_quantization/tensorflow/mobilenet_v2/main.py:152]: Good question! I hope you will find answer in ref 12610 and https://docs.openvino.ai/2023.2/quantization_w_accuracy_control.html

File: examples/post_training_quantization/torch/mobilenet_v2/main.py (hunk 2 above)
Comment (alexsu52) [reply examples/post_training_quantization/torch/mobilenet_v2/main.py:144]: Thanks for the comment! I submitted new reference metrics.

File: examples/post_training_quantization/tensorflow/mobilenet_v2/main.py (hunk 4 above)
Comment (alexsu52) [reply examples/post_training_quantization/tensorflow/mobilenet_v2/main.py:149]: Good question! openvino.convert_model works incorrect for  share_weights=True (ref 126201)

File: examples/post_training_quantization/tensorflow/mobilenet_v2/main.py (hunk 1 above)
Comment (KodiaqQ) [reply examples/post_training_quantization/tensorflow/mobilenet_v2/main.py:152]: I see and understand the reason. Does it change mean that `ov.serialize` was able to save the model in the FP32 format without any additional flags before? Or it was a bug in our example that the model was converted into FP16 previously?

**Review by KodiaqQ — APPROVED** [review]

**Review by nikita-savelyevv — APPROVED** [review]
